pkg/server/handler.go

```diff
@@ -128,7 +137,22 @@ func Get(w http.ResponseWriter, state *terraform.State, store storage.Storage, k
 	HTTPResponse(w, http.StatusOK, string(state.Data))
 }
 
-func Post(w http.ResponseWriter, state *terraform.State, body []byte, store storage.Storage, kms kms.KMS) {
+func Post(r *http.Request, w http.ResponseWriter, state *terraform.State, body []byte, locker lock.Locker, store storage.Storage, kms kms.KMS) {
+	reqLockID := r.URL.Query().Get("ID")
+
+	lock, err := locker.GetLock(state)
+	if err != nil {
+		log.Errorf("failed to get lock for state with id %s: %v", state.ID, err)
+		HTTPResponse(w, http.StatusInternalServerError, "")
```

Comment:
Is this really a server error? I guess, most likely the Terraform user disabled locking.

Reply:
You are right, I changed it to Warn in accordance with other log calls.

Reply:
In my opinion the HTTP status code 500 also should be changed to "Bad Request" or something similar.

Reply:
I agree and changed two similar occasions.